cfgov/unprocessed/js/organisms/MegaMenuMobile.js

```diff
@@ -92,6 +86,58 @@ function MegaMenuMobile( menus ) {
   }
 
   /**
+   * Event handler for when FlyoutMenu trigger is clicked.
+   * @param {Event} event - A FlyoutMenu event.
+   */
+  function _handleTriggerClick( event ) {
+    this.dispatchEvent( 'triggerClick', { target: this } );
+    var menu = event.target;
+    var rootMenu = _menus.getRoot().data;
+    var menuNode = menu.getData();
+    var level = menuNode.level;
+    var transition = rootMenu.getTransition();
+
+    if ( menu === rootMenu ) {
+      // Root menu clicked.
+
+      // Root menu is closing.
+      if ( menu.isExpanded() ) {
+        level = _activeMenu.getData().level;
+        menu.setCollapseTransition( transition,
+                                    transition.moveLeft, [ level + 1 ] );
+      }
+    } else {
+      // Submenu clicked.
+      var siblings = _menus.getAllAtLevel( level );
+      var siblingMenu;
+      for ( var i = 0, len = siblings.length; i < len; i++ ) {
+        siblingMenu = siblings[i].data;
+        siblingMenu
+          .setExpandTransition( transition, transition.moveLeft, [ level ] );
+        // If on the 2nd level menu, set the back button to moveToOrigin,
+        // otherwise we're on the 3rd level menu, so moveLeft is needed.
+        if ( level === 1 ) {
+          siblingMenu
+            .setCollapseTransition( transition, transition.moveToOrigin );
+        } else {
+          siblingMenu.setCollapseTransition( transition, transition.moveLeft );
+        }
+        // If we're on the current menu, show it & hide all the other siblings.
+        if ( siblings[i] === menuNode ) {
+          siblingMenu.getDom().content.classList.remove( 'u-invisible' );
+        } else {
+          siblingMenu.getDom().content.classList.add( 'u-invisible' );
+        }
+      }
+
+      // TODO: Investigate helper functions to mask these crazy long lookups!
```

Comment:
Yeah, so the setup is there's the data that's in the data model tree, which allows walking around the menu without touching the dom, and each tree node contains a `data` reference to a menu, which allows access to the dom. Then in the reverse, the menus contain a reference to the tree node (through the `setData` and `getData` methods on `FlyoutMenu`). This allows looking up and moving around the data model tree from an event handler attached to the dom. Best will likely be replacing these long lookup chains with something like `dom2tree()` and `tree2dom()` methods that translate back and forth between the two as needed.